trestle/tasks/osco_to_oscal.py

```diff
@@ -119,33 +120,30 @@ def simulate(self) -> TaskOutcome:
-                # calculate the output file, including path
-                ofile = opth / pathlib.Path(ifile.stem+'.json')
-                # only allow writing output file if either:
-                # a) it does not already exist, or
-                # b) output-overwrite flag is True
-                if not overwrite:
-                    if ofile.exists():
-                        logger.error(f'simluate: file exists: {ofile}')
-                        return TaskOutcome('simulated-failure')
-                if not quiet:
-                    logger.debug(f'[simluate]  create {ofile}')
-                # fetch the contents of the subject OSCO .yaml/.yml file
-                idata = self._read_content(ifile)
-                # create the OSCAL .json file from the OSCO and the optional osco-metadata files
-                observations, analysis = osco.get_observations(idata, metadata)
-                # write the OSCAL to the output file
-                self._write_content(ofile, observations, True)
-                # display analysis
-                if not quiet:
-                    logger.debug(f'[simluate] Rules Analysis:')
-                    logger.debug(f'[simluate] config_maps: {analysis["config_maps"]}')
-                    logger.debug(f'[simluate] dispatched rules: {analysis["dispatched_rules"]}')
-                    logger.debug(f'[simluate] result types: {analysis["result_types"]}')
+                # assemble collection comprising output file name to unprocessed content
+                collection = self._assemble(ifile)
+                # formulate each output OSCAL partial results file
+                for oname in collection.keys():
+                    ofile = opth / pathlib.Path(oname)
+                    # only allow writing output file if either:
+                    # a) it does not already exist, or
+                    # b) output-overwrite flag is True
+                    if not overwrite:
+                        if ofile.exists():
+                            logger.error(f'file exists: {ofile}')
+                            return TaskOutcome('simulated-failure')
+                    if not quiet:
+                        logger.debug(f'create: {ofile}')
+                    # create the OSCAL .json file from the OSCO and the optional osco-metadata files
+                    observations, analysis = osco.get_observations(collection[oname], metadata)
+                    # write the OSCAL to the output file
+                    self._write_content(ofile, observations, True)
+                    # display analysis
+                    if not quiet:
+                        logger.debug(f'[simluate] Rules Analysis:')
+                        logger.debug(f'[simluate] config_maps: {analysis["config_maps"]}')
+                        logger.debug(f'[simluate] dispatched rules: {analysis["dispatched_rules"]}')
+                        logger.debug(f'[simluate] result types: {analysis["result_types"]}')
```

Comment:
So one item I forgot to mention. The logger should automatically observe a -v flag passed into the CLI which will turn on debug logging. If that behaves as expected that would be my preference (as it removes the need for an additional flag.

It is not a blocker more of a FYI

Reply:
For the most part the code does expect and follow the standard of the -v flag. Note that the task produces logging output about the file written and some analysis of the results encountered at the info level of execute. The quiet flag is meant to suppress all logging output, meaning the aforementioned info. At this point it is a frill, you might say, and can be removed entirely...or not. Nevertheless and with respect to symmetry, for the case of simulate these outputs are at the debug level while at the execute level they are at the info level. I think it's OK the way it is, but am not wedded to the need for the quiet flag. If preference is to remove, just say the word. Is there a way to suppress .info level output already?